File: python/migration.py
import requests
from requests_oauthlib import OAuth1
import json
from QBO_request import api_call

with open('config.json', 'r') as f:
    config = json.load(f)

def migrate_tokens():
    '''Migrate tokens from OAuth1 to OAuth2'''
    headers = { 'Content-Type': 'application/json' }
    auth = OAuth1 (
        config['OAuth1']['consumer_key'], 
        config['OAuth1']['consumer_secret'],
        config['OAuth1']['access_token'], 
        config['OAuth1']['access_secret']
    )
    body = {
        'scope':'com.intuit.quickbooks.accounting',
        'redirect_uri': config['Redirect_url'],
        'client_id': config['OAuth2ClientId'],
        'client_secret': config['OAuth2ClientSecret']
    }
    r = requests.post(config['Migration_url'], headers=headers, auth=auth, data=json.dumps(body))
    print ('Migration API call:')
    print ('Status code: '+str(r.status_code))
    print (r.json())
    return r.json()

oauth2_tokens = migrate_tokens()

# Optionally, make a sample QBO API request
company_info = api_call(oauth2_tokens['access_token'], oauth2_tokens['realmId'])

File: ruby/migration.rb
require 'oauth/request_proxy/typhoeus_request'
require 'oauth'
require 'json'
require 'yaml'
require_relative 'QBO_request'

def migrate_tokens()
	@consumer = OAuth::Consumer.new(
		CONFIG['consumer_key'],
		CONFIG['consumer_secret'],
	)
	@token = OAuth::Token.new(
		CONFIG['access_token'], 
		CONFIG['access_secret']
	)
	
	options = {
		method: :post,
		headers: { 'Content-Type' => 'application/json' },
		body: {
        	scope:'com.intuit.quickbooks.accounting',
        	redirect_uri: CONFIG['redirect_uri'],
        	client_id: CONFIG['oauth2_client_id'],
        	client_secret: CONFIG['oauth2_client_secret']

    	}.to_json
	}

	oauth_params = {:consumer => @consumer, :token => @token}

	req = Typhoeus::Request.new(CONFIG['migration_url'], options)
	oauth_helper = OAuth::Client::Helper.new(req, oauth_params.merge(:request_uri => CONFIG['migration_url']))
	req.options[:headers].merge!({
		'Authorization' => oauth_helper.header,
    })
	req.run
	response = req.response
	puts 'Migrate API call'
	puts response.code 
	puts response.headers 
	puts response.body 
	JSON.parse(response.body)
end

oauth2_tokens = migrate_tokens()

# Optionally, make a sample QBO API request
api_call(oauth2_tokens['access_token'], oauth2_tokens['realmId'])